velox/vector/DecodedVector.h

```diff
@@ -209,6 +209,16 @@ class DecodedVector {
     return bits::isBitNull(nulls_, indices_[idx]);
   }
 
+  /// Return whether this top-level vector has the same value at the given index
+  /// as the other top-level vector at the other vector's index.
+  bool equalValueAt(
+      const DecodedVector& other,
+      vector_size_t idx,
+      vector_size_t otherIndex) const {
+    return baseVector_->equalValueAt(
```

Comment:
Comparison of complex types requires to check for nested nulls and throw. I think we need to add CompareFlags parameter to this API to allow for that.